packages/component-base/src/virtualizer-iron-list-adapter.js

```diff
@@ -281,6 +281,19 @@ export class IronListAdapter {
       this._virtualCount = this.items.length;
     }
 
+    // When reducing size while invisible, iron-list does not update items, so
+    // their hidden state is not updated and their __lastUpdatedIndex is not
+    // reset. In that case do it manually here.
+    if (!this._visible) {
+      this._iterateItems((pidx, vidx) => {
+        const el = this._physicalItems[pidx];
+        el.hidden = vidx >= size;
+        if (el.hidden) {
+          delete el.__lastUpdatedIndex;
+        }
+      });
```

Comment:
Could we call `this._assignModels()` here instead?

Reply:
We could, but that would also update items that are not hidden based on their index, which seemed unnecessary given that the container is invisible. I can change it if you think it's preferable.

Reply:
> that would also update items that are not hidden based on their index

I'd assume the non-hidden elements would not get updated if their index hasn't changed due to the check [here](https://github.com/vaadin/web-components/blob/1f6e834ebc59322de3ab0baf3c7445415d4ffffc/packages/component-base/src/virtualizer-iron-list-adapter.js#L206).

Reply:
Makes sense, updated to use `_assignModels`.